signer/src/stacks/wallet.rs

```diff
@@ -0,0 +1,537 @@
+//! This module contains functionality for signing stacks transactions
+//! using the signers' multi-sig wallet.
+
+use std::collections::BTreeMap;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering;
+
+use blockstack_lib::address::C32_ADDRESS_VERSION_MAINNET_MULTISIG;
+use blockstack_lib::address::C32_ADDRESS_VERSION_TESTNET_MULTISIG;
+use blockstack_lib::chainstate::stacks::OrderIndependentMultisigHashMode;
+use blockstack_lib::chainstate::stacks::OrderIndependentMultisigSpendingCondition;
+use blockstack_lib::chainstate::stacks::StacksTransaction;
+use blockstack_lib::chainstate::stacks::TransactionAnchorMode;
+use blockstack_lib::chainstate::stacks::TransactionAuth;
+use blockstack_lib::chainstate::stacks::TransactionAuthFlags;
+use blockstack_lib::chainstate::stacks::TransactionPayload;
+use blockstack_lib::chainstate::stacks::TransactionPublicKeyEncoding;
+use blockstack_lib::chainstate::stacks::TransactionSpendingCondition;
+use blockstack_lib::chainstate::stacks::TransactionVersion;
+use blockstack_lib::core::CHAIN_ID_MAINNET;
+use blockstack_lib::core::CHAIN_ID_TESTNET;
+use blockstack_lib::types::chainstate::StacksAddress;
+use blockstack_lib::util::secp256k1::MessageSignature;
+use blockstack_lib::util::secp256k1::Secp256k1PublicKey;
+use secp256k1::ecdsa::RecoverableSignature;
+use secp256k1::Message;
+use secp256k1::PublicKey;
+use secp256k1::SecretKey;
+use secp256k1::SECP256K1;
+
+use crate::config::NetworkKind;
+use crate::error::Error;
+use crate::stacks::contracts::AsContractCall;
+
+/// Requisite info for the signers' multi-sig wallet on Stacks.
+#[derive(Debug, Clone)]
+pub struct SignerWallet {
+    /// The current set of public keys for all known signers during this
+    /// PoX cycle. These values must be sorted.
+    public_keys: Vec<PublicKey>,
```

Comment:
How about using a `BTreeSet` instead of a `Vec` to enforce that the values are sorted?

Reply:
That makes sense, will update.